selftest: mptcp: cover the SimultConnectFallback

Dwyane-Yan opened the pull request:
This patch adds a test that cover simultaneous initiation. In this situation, the msk should fallback to TCP, and this patch can check the fallback behavior.

Dwyane-Yan committed:
selftest: mptcp: cover the SimultConnectFallback

This patch adds a test that cover simultaneous initiation. In this
situation, the msk should fallback to TCP, and this patch can check the
fallback behavior.

Suggested-by: Matthieu Baerts (NGI0) <matttbe@kernel.org>
Signed-off-by: Gang Yan <yangang@kylinos.cn>

matttbe commented on gtests/net/mptcp/syscalls/client_selfconnect.pkt:
```diff
@@ -0,0 +1,19 @@
++0     <  S  0:0(0)         win 1000   <mss 1460, sackOK, TS val 407 ecr 0,         nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
++0     >  S. 0:0(0)  ack 1             <mss 1460, sackOK, TS val 330 ecr 0,         nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
++0     <  S. 0:0(0)  ack 1  win 65535  <mss 1460, sackOK, TS val 700 ecr 100,       nop, wscale 8, mpcapable v1 flags[flag_h] key[skey=2]>
++0     >   . 1:1(0)  ack 1             <nop,      nop,    TS val 845707014 ecr 700, nop, nop,      sack 0:1>
```
why do you have such a huge value for the "TS val"? Can you not have something smaller in 3 digits, as long as it is bigger than 330?

matttbe commented on gtests/net/mptcp/syscalls/client_selfconnect.pkt:
```diff
@@ -0,0 +1,19 @@
++0     <  S. 0:0(0)  ack 1  win 65535  <mss 1460, sackOK, TS val 700 ecr 100,       nop, wscale 8, mpcapable v1 flags[flag_h] key[skey=2]>
++0     >   . 1:1(0)  ack 1             <nop,      nop,    TS val 845707014 ecr 700, nop, nop,      sack 0:1>
+
+//check for fallback
```
(detail: clearer to add a space after `//`)

Good idea to check for fallbacks!

matttbe commented on gtests/net/mptcp/syscalls/client_selfconnect.pkt:
```diff
@@ -0,0 +1,19 @@
+// connect() function, a client tries to connect to itself
```
It is not about connecting to itself, but about simultaneous initiation. Please see the TCP RFC for more details: https://www.ietf.org/rfc/rfc9293.html#section-3.5-7

In the test, it is another peer that is injecting the SYN. Don't hesitate to take a packet capture when running the tests.

Can you then update this comment, and the commit message to mention that it is about "simultaneous connect", and not "connecting to itself"?

matttbe reviewed (changes requested):
Thank you for looking at this! I mainly have one comment about the description.

Dwyane-Yan commented:
Hi Matt, 
Thanks for your reminder. 
I've reviewed the materials on Simultaneous Connection and TS val/ecr,  and made the following revisions: 
1. Adjusted some TS val/ecr values; 
2. Renamed the file to 'simult_connect.pkt'; 
3.  Updated the comment information.

matttbe reviewed (approved):
thank you for the update!
